JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,329 @@
+	           	BufferedInputStream in = new BufferedInputStream(connection.getInputStream());
+	    	    ByteArrayOutputStream buf = new ByteArrayOutputStream();
+	            readResult(in, buf);
+	    	    String res = buf.toString("UTF-8");
+	        	JsonElement jElement = JsonParser.parseString(res);
+	            JsonObject jObject = jElement.getAsJsonObject();
+				// set Tokens as System Properties - new values to access_token and refresh_token
+	            String accessToken = jObject.has("access_token") ? jObject.get("access_token").getAsString() : null;
+	            String refreshToken = jObject.has("refresh_token") ? jObject.get("refresh_token").getAsString() : null;           
+	            if (null == accessToken || null == refreshToken)
+	            {
+	            	throw new Exception("Access/refresh token is null, Please verify the config.properties for proper inputs.");
+	            }
+				System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, accessToken);
+				System.setProperty(OAUTH_REFRESH_TOKEN_VAR_STRING, refreshToken);
+	        }catch(UnsupportedEncodingException uee)
+			{
+				uee.printStackTrace();
+	        } catch (Exception e) 
+	        {
+	            String res = "";
+	            try 
+	            {
+	                BufferedInputStream in = new BufferedInputStream(connection.getErrorStream());
+	                ByteArrayOutputStream buf = new ByteArrayOutputStream();
+	                readResult(in, buf);
+	                res = buf.toString("UTF-8");
+	            } catch (Exception ex) 
+	            {
+	                //Improper setup. Passes in SF, fails in Devjail. Skip when this happens, but print the error.
+	                ex.printStackTrace();
+	            }
+				throw e;
+	        } finally 
+	        {
+	            connection.disconnect();
+	        }
+    	}catch (IOException unreportedex) 
+    	{
+    		unreportedex.printStackTrace();
         }
     }
-
-    @Override
-    public Integer call() throws Exception {
-        try {
-            Connection conn = connectToDB(host, port, dbName, accessToken, refreshToken, clientSecret);
-            ResultSet rs = executeQuery(conn);
-            printResults(rs);
-            conn.close();
-        } catch (SQLTransientConnectionException connException) {
-            System.out.print("Network connection issue: ");
-            System.out.print(connException.getMessage());
-            System.out.println(" Try again later!");
-        } catch (SQLInvalidAuthorizationSpecException authException) {
-            System.out.print("Could not log into database: ");
-            System.out.print(authException.getMessage());
-            System.out.println(" Check the login credentials and try again.");
-        } catch (SQLException e) {
+	// if no access token is set, obtains and sets first access and refresh tokens
+	// uses the password grant flow
+	private static void EnsureAccessToken() throws Exception 
```

Comment:
The EnsureAccessToken() method in the ADO.NET example has been updated to show the correct flow.